src/reducers/app.js

```diff
@@ -245,6 +246,22 @@ const eventDelayTracks: Reducer<boolean> = (state = false, action) => {
   }
 };
 
+/**
+ * Holds the state for whether the user has the local data to delete the
+ * currently displayed profile.
+ */
+const cachedStoredProfileData: Reducer<ProfileData | null> = (
```

Comment:
As part of the global state, I think this could use a bit more precise of a name. It's not clear what is going on here. This also ties into my comment on renaming the `ProfileData` type.

I'm also suspicous that this won't be correctly invalidated when viewing a different profile. I can't confirm that with the current patch, as I don't know how it's being used.

e.g. in `src/reducers/profile-view.js`
```
/**
 * Provide a mechanism to wrap the reducer in a special function that can reset
 * the state to the default values. This is useful when viewing multiple profiles
 * (e.g. in zip files).
 */
const wrapReducerInResetter = (
  regularReducer: Reducer<ProfileViewState>
): Reducer<ProfileViewState> => {
  return (state, action) => {
    switch (action.type) {
      case 'SANITIZED_PROFILE_PUBLISHED':
      case 'REVERT_TO_PRE_PUBLISHED_STATE':
      case 'RETURN_TO_ZIP_FILE_LIST':
        // Provide a mechanism to wipe the state clean when changing out profiles.
        // All of the profile view information is invalidated.
        return regularReducer(undefined, action);
      default:
        // Run the normal reducer.
        return regularReducer(state, action);
    }
  };
};
```

Reply:
Currently it is managed by LoadStoredProfileDataManager only, therefore it's tied to a hash. It has no meaning outside of uploaded profiles anyway, so it has no meaning with zip files especially. When publishing a new profile especially the hash will change and therefore the component will load the new data into this state.

But I realize that I don't clear this state when the hash moves to a value that's not stored in the indexed db, I'll make sure this is fixed and tested. Also I'll add a few comments here to explain this a bit.

Reply:
> But I realize that I don't clear this state when the hash moves to a value that's not stored in the indexed db, I'll make sure this is fixed and tested. Also I'll add a few comments here to explain this a bit.

This is part is fixed.

Reply:
> As part of the global state, I think this could use a bit more precise of a name.

I used `currentProfileUploadedInformation`, I think this is better indeed!